avocado/core/task/statemachine.py

```diff
@@ -188,6 +193,24 @@ async def triage(self):
                 LOG.debug('Task "%s" has failed dependencies',
                           runtime_task.task.identifier)
                 return
+        if runtime_task.task.category != "test":
+            async with self._state_machine.cache_lock:
+                is_task_in_cache = await self._spawner.is_requirement_in_cache(
+                    runtime_task)
+                if is_task_in_cache is None:
+                    async with self._state_machine.lock:
+                        self._state_machine.triaging.append(runtime_task)
+                        runtime_task.status = 'WAITING'
+                        await asyncio.sleep(0.1)
+                    return
+
+                if is_task_in_cache:
+                    await self._state_machine.finish_task(
+                        runtime_task, "FINISHED: Task in cache")
```

Comment:
I think `FINISHED` is not needed, given that `finish_test` will already put that information in the log.

Reply:
`FINISHED` is important here because it will be part of `runtime_task` status and the status is used for detecting all finished tasks in [here](https://github.com/avocado-framework/avocado/pull/5412/files#diff-b2cc6571a0856868fe60a323f79a542765b0fa23da3d10876423e3d6de73e7fcR68).